### PR TITLE
drivers: watchdog: litex: add litex watchdog

### PR DESCRIPTION
**Pull request by maass-hamburg**
this adds a driver for the litex watchdog.

on LiteX: https://github.com/enjoy-digital/litex/pull/1996

### CODE REVIEW

**Comment by fkokosinski** [comment]
Hey, thanks for this PR. In general the code LGTM.

That being said, the watchdog driver is not being used by the `litex_vexriscv` platform (or any other in-tree platform), meaning it will not be built/tested by twister in the CI. This essentially means that from the CI standpoint this is dead code.

Could you perhaps add it to the `litex_vexriscv` platform and do the RTL configuration repo that this platform is targeting? Thanks

File: drivers/interrupt_controller/intc_vexriscv_litex.c
Comment (fkokosinski) [inline drivers/interrupt_controller/intc_vexriscv_litex.c:61]: The name of this macro is somewhat ambiguous

File: drivers/interrupt_controller/intc_vexriscv_litex.c
Comment (fkokosinski) [inline drivers/interrupt_controller/intc_vexriscv_litex.c:114]: Could you apply a similar change to other IRQs handled in this routine?

**Comment by maass-hamburg** [comment]
> RTL configuration repo

It is now added to the `litex_vexriscv` platform. I also added the watchdog here: https://github.com/litex-hub/zephyr-on-litex-vexriscv/pull/21

File: drivers/interrupt_controller/intc_vexriscv_litex.c
Comment (maass-hamburg) [reply drivers/interrupt_controller/intc_vexriscv_litex.c]: changed it to use `DT_FOREACH_STATUS_OKAY_NODE`. That way the interrupt controller doesn't has to be modified anymore, when new peripherals are added. This way we can also support custom out off tree peripherals and drivers with interrupts without having to edit this.

File: drivers/watchdog/wdt_litex.c
Comment (ycsin) [inline drivers/watchdog/wdt_litex.c:210]:
```suggestion
	.feed = wdt_litex_feed,
```

File: drivers/interrupt_controller/intc_vexriscv_litex.c
Comment (maass-hamburg) [reply drivers/interrupt_controller/intc_vexriscv_litex.c:57]: Litex only has one interrupt controller and this interrupt controller driver also only supports one instance. There shouldn't be any other interrupt controller.

File: drivers/watchdog/wdt_litex.c
Comment (fkokosinski) [inline drivers/watchdog/wdt_litex.c:2]: The copyright seems to be wrong

Comment (maass-hamburg) [reply drivers/watchdog/wdt_litex.c]: Good catch ;)

**Comment by maass-hamburg** [comment]
ping

**Review by fkokosinski — DISMISSED** [review]

**Comment by maass-hamburg** [comment]
ping @tgorochowik @kgugala

**Comment by maass-hamburg** [comment]
litte reminder for the second review @tgorochowik @kgugala

**Review by tgorochowik — DISMISSED** [review]

**Comment by carlescufi** [comment]
@maass-hamburg please rebase

**Comment by maass-hamburg** [comment]
rebase done to resolve conflicts

**Review by tgorochowik — DISMISSED** [review]

**Comment by maass-hamburg** [comment]
a `};` got lost in the dtsi during rebase

**Comment by maass-hamburg** [comment]
@fkokosinski @tgorochowik litte reminder to re-approve after the rebase

**Review by fkokosinski — DISMISSED** [review]

File: drivers/watchdog/wdt_litex.c
Comment (tgorochowik) [inline drivers/watchdog/wdt_litex.c:73]: was this tested? You seem to be using bits as size in the corresponding litex PR: https://github.com/enjoy-digital/litex/pull/1996/files#diff-91072e8d1728a622115f31f10d4a9a6902f27f22054a6236b4b09e00aa97287aR23

Or is it converted to bytes in litex somewhere else

Comment (maass-hamburg) [reply drivers/watchdog/wdt_litex.c:73]: It is the size of the reg in the dts, which is in bytes

Comment (tgorochowik) [reply drivers/watchdog/wdt_litex.c:73]: oh, right

**Review by tgorochowik — DISMISSED** [review]

**Comment by aescolar** [comment]
@maass-hamburg  rebase needed

**Comment by maass-hamburg** [comment]
Rebased to resolve conflicts

**Review by tgorochowik — APPROVED** [review]

**Review by fkokosinski — APPROVED** [review]

**Pull request merged**
